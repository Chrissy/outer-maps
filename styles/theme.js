import spacingScale from "./spacingScale";
import typeScale from "./typeScale";

export default {
  brandColor: "#344632",
  accentColor: "#FF6000",
  accentColorTintLighter: "#FFAE86",
  accentColorTintLight: "#FF8244",
  accentColorTintDark: "#CA4300",
  accentColorTintDarker: "#9B3300",
  gray1: "#F7F7F7",
  gray2: "#F3F3F3",
  gray3: "#E9E9E9",
  gray4: "#D8D8D8",
  gray5: "#A7A7A7",
  gray6: "#838383",
  gray7: "#494949",
  gray8: "#292929",
  blue: "#3232FF",
  purple: "#5700EC",
  orange: "#FF9500",
  red: "#D0011B",
  ss: spacingScale,
  ts: typeScale,
  headlineFont: "\"Hoefler Text A\", \"Hoefler Text B\"",
  bodyFont:
    "\"Gotham Narrow A\", \"Gotham Narrow B\", -apple-system, BlinkMacSystemFont, \"Segoe UI\", Roboto, Helvetica, Arial, sans-serif, \"Apple Color Emoji\", \"Segoe UI Emoji\", \"Segoe UI Symbol\""
};
