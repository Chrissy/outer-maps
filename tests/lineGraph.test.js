import React from "react";
import LineGraph from "../components/lineGraph.js";
import renderer from "react-test-renderer";
import theme from "../styles/theme";
import { ThemeProvider } from "emotion-theming";

const elevations = [
  {
    distanceFromPreviousPoint: 1,
    elevation: 1000
  },
  {
    distanceFromPreviousPoint: 5,
    elevation: 10000
  },
  {
    distanceFromPreviousPoint: 1000,
    elevation: 33
  }
];

it("renders", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <LineGraph elevations={elevations} colors={["#555"]} />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
