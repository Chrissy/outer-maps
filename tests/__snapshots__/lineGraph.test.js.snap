// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.emotion-10 {
  margin-top: 0.25em;
  position: relative;
  width: 100%;
  line-height: 0;
  padding: 0 1em;
  box-sizing: border-box;
}

.emotion-2 {
  width: 100%;
}

.emotion-0 {
  fill: #FF5500;
  stroke: #FF5500;
  stroke-width: 1px;
}

.emotion-8 {
  position: absolute;
  top: 0;
  left: 0;
  width: calc(100% - 2em);
  height: 100%;
  margin: 0 1em;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5800EC;
  padding: 0 1.25em;
  font-size: 0.5em;
  font-weight: 700;
  position: absolute;
  left: 0%;
  top: 67.72348750877897%;
  -webkit-transform: translate(-30%,35%);
  -ms-transform: translate(-30%,35%);
  transform: translate(-30%,35%);
  border-radius: 1.5em;
  min-width: 25px;
  color: #fff;
  height: 1.75em;
  border: 2px solid #fff;
}

.emotion-4:after {
  display: block;
  content: "";
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #5800EC;
  position: absolute;
  left: 30%;
  top: 100%;
  -webkit-transform: translate(-50%,-10%);
  -ms-transform: translate(-50%,-10%);
  transform: translate(-50%,-10%);
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5800EC;
  padding: 0 1.25em;
  font-size: 0.5em;
  font-weight: 700;
  position: absolute;
  left: 100%;
  top: 75%;
  -webkit-transform: translate(-70%,35%);
  -ms-transform: translate(-70%,35%);
  transform: translate(-70%,35%);
  border-radius: 1.5em;
  min-width: 25px;
  color: #fff;
  height: 1.75em;
  border: 2px solid #fff;
}

.emotion-6:after {
  display: block;
  content: "";
  width: 0;
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #5800EC;
  position: absolute;
  left: 70%;
  top: 100%;
  -webkit-transform: translate(-50%,-10%);
  -ms-transform: translate(-50%,-10%);
  transform: translate(-50%,-10%);
}

<div
  className="emotion-10 emotion-11"
>
  <svg
    className="emotion-2 emotion-3"
    viewBox="0 0 275 110"
  >
    <g />
    <polyline
      className="emotion-0 emotion-1"
      points="0,110 0,99.49583625965687 1.3681592039800994,25 275,107.5 275,110"
    />
  </svg>
  <div
    className="emotion-8 emotion-9"
  >
    <div
      className="emotion-4 emotion-5"
      x={0}
      y={67.72348750877897}
    >
      1,000
    </div>
    <div
      className="emotion-6 emotion-5"
      x={100}
      y={75}
    >
      33
    </div>
  </div>
</div>
`;
